frappe/public/js/integrations/google_drive_picker.js

```diff
@@ -1,85 +1,85 @@
+			});
 		}
 
-		gapi.load("picker", this.onPickerApiLoad.bind(this));
+		function ajax_gapi() {
+			// load the google API library
+			$.ajax({
+				method: "GET",
+				url: "https://apis.google.com/js/api.js",
+				dataType: "script",
+				cache: true,
+			});
+		}
 	}
-
-	onAuthApiLoad() {
-		gapi.auth.authorize(
-			{
-				client_id: this.clientId,
-				scope: this.scope,
-				immediate: false,
+	libsLoaded() {
+		this.tokenClient = google.accounts.oauth2.initTokenClient({
+			client_id: this.clientId,
+			scope: this.scope,
+			callback: async (response) => {
+				if (response.error !== undefined) {
+					frappe.throw(response);
+				}
+				this.accessToken = response.access_token;
+				frappe.boot.user.google_drive_token = response.access_token;
+				await this.createPicker();
 			},
-			this.handleAuthResult.bind(this)
-		);
-	}
-
-	handleAuthResult(authResult) {
-		let error_map = {
-			popup_closed_by_user: __("Google Authentication was closed abruptly by the user"),
-		};
-
-		if (authResult && !authResult.error) {
-			frappe.boot.user.google_drive_token = authResult.access_token;
-			this.createPicker();
+		});
+		this.gisInited = true;
+		gapi.load("client:picker", { callback: this.initializePicker.bind(this) });
+		if (frappe.boot.user.google_drive_token === null) {
+			// Prompt the user to select a Google Account and ask for consent to share their data
+			// when establishing a new session.
+			this.tokenClient.requestAccessToken({ prompt: "consent" });
 		} else {
-			let error = error_map[authResult.error] || __("Google Authentication Error");
-			frappe.throw(error);
+			// Skip display of account chooser and consent dialog for an existing session.
+			this.tokenClient.requestAccessToken({ prompt: "" });
 		}
 	}
-
-	onPickerApiLoad() {
-		this.pickerApiLoaded = true;
-		this.createPicker();
+	async initializePicker() {
+		gapi.client.load("https://www.googleapis.com/discovery/v1/apis/drive/v3/rest");
+		this.pickerInited = true;
 	}
-
 	createPicker() {
-		// Create and render a Picker object for searching images.
-		if (this.pickerApiLoaded && frappe.boot.user.google_drive_token) {
-			this.view = new google.picker.DocsView(google.picker.ViewId.DOCS)
-				.setParent("root") // show the root folder by default
-				.setIncludeFolders(true); // also show folders, not just files
-
-			this.picker = new google.picker.PickerBuilder()
-				.setAppId(this.appId)
-				.setDeveloperKey(this.developerKey)
-				.setOAuthToken(frappe.boot.user.google_drive_token)
-				.addView(this.view)
-				.setLocale(frappe.boot.lang)
-				.setCallback(this.pickerCallback)
-				.build();
-
-			this.picker.setVisible(true);
-			this.setupHide();
-		}
+		this.view = new google.picker.View(google.picker.ViewId.DOCS);
+		this.view.setMimeTypes("image/png,image/jpeg,image/jpg");
+		this.picker = new google.picker.PickerBuilder()
+			.enableFeature(google.picker.Feature.NAV_HIDDEN)
+			.enableFeature(google.picker.Feature.MULTISELECT_ENABLED)
```

Comment:
```suggestion
```

Multiselect is not supported by frappe yet. If you want to enable it here, you'll have to take care of the rest of file handling as well.